### PR TITLE
ENH: Add CLI to detect usable estimators within a BIDS dataset

### PR DESCRIPTION
**Pull request by mgxd**
Quick stab at this, still need to add some tests

### CODE REVIEW

**Comment by mgxd** [comment]
@oesteban @effigies any chance this can get a review so we can sneak it into this release?

nvm, this will have to wait for a new niworkflows release.

**Comment by mgxd** [comment]
Okay, I'm satisfied with this now.

Sample usage / output:
```
$ sdcflows-find-estimators ~/code/fmriprep/testing/ds054
Estimation for </Users/mathiasg/code/fmriprep/testing/ds054> complete. Found:
	sub-100185
		FieldmapEstimation(sources=<3 files>, method=<EstimatorType.PHASEDIFF: 3>, bids_id='auto_00000')
			i-	fmap/sub-100185_phasediff.nii.gz
			i-	fmap/sub-100185_magnitude1.nii.gz
			i-	fmap/sub-100185_magnitude2.nii.gz
```

File: sdcflows/cli/find_estimators.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    _indexer = BIDSLayoutIndexer(
+        validate=False,
+        ignore=(
+            "code",
+            "stimuli",
+            "sourcedata",
+            "models",
+            "derivatives",
+            re.compile(r"^\."),
+            re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(beh|eeg|ieeg|meg|perf)"),
```
Comment (effigies) [inline sdcflows/cli/find_estimators.py]: Might as well ignore these, too.

```suggestion
            re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(anat|beh|eeg|ieeg|meg|micr|perf)"),
```

**Review by effigies — APPROVED** [review]
LGTM!

File: sdcflows/cli/find_estimators.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    _indexer = BIDSLayoutIndexer(
+        validate=False,
+        ignore=(
+            "code",
+            "stimuli",
+            "sourcedata",
+            "models",
+            "derivatives",
+            re.compile(r"^\."),
+            re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(anat|beh|eeg|ieeg|meg|micr|perf)"),
```
Comment (effigies) [inline sdcflows/cli/find_estimators.py]: Realizing we're going to need anat for SyN.

```suggestion
            re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(beh|eeg|ieeg|meg|micr|perf)"),
```

**Pull request merged**